lib/engine/game/g_18_ga/game.rb

```diff
@@ -193,6 +203,47 @@ def event_close_companies!
         def remove_icon_from_waycross
           waycross_hex.tile.icons = []
         end
+
+        def georgia_railroad
+          corporation_by_id('GA')
+        end
+
+        def savannah
+          @savannah ||= hex_by_id('G13')
+        end
+
+        def charleston
+          @charleston ||= hex_by_id('E15')
+        end
+
+        def optional_hexes
+          return self.class::HEXES unless @optional_rules&.include?(:cotton_port)
+
+          new_hexes = {}
+          HEXES.keys.each do |color|
```

Comment:
```suggestion
          HEXES.each_key do |color|
```

Reply:
done, though I'll note that every other instance of this type of code on the site uses HEXES.keys.each.

this works too, so I'll go with it.

Reply:
> done, though I'll note that every other instance of this type of code on the site uses HEXES.keys.each.

All the other code is wrong ;-) Using Hash.keys.each (at least in Opal) creates a new array containing a copy of the hash's keys, whereas Hash.each_key directly uses the hash to iterate, without needing to create a new array.